packages/babel-helper-optimise-call-expression/src/index.ts

```diff
@@ -1,20 +1,26 @@
 import * as t from "@babel/types";
+import type {
+  CallExpression,
+  Expression,
+  OptionalCallExpression,
+  SpreadElement,
+} from "@babel/types";
 
 /**
  * A helper function that generates a new call expression with given thisNode.
  It will also optimize `(...arguments)` to `.apply(arguments)`
  *
  * @export
- * @param {Node} callee The callee of call expression
- * @param {Node} thisNode The desired this of call expression
- * @param {Node[]} args The arguments of call expression
+ * @param {Expression} callee The callee of call expression
+ * @param {Expression} thisNode The desired this of call expression
+ * @param {Readonly<Array<Expression | SpreadElement>>} args The arguments of call expression
  * @param {boolean} optional Whether the call expression is optional
  * @returns {CallExpression | OptionalCallExpression} The generated new call expression
  */
-export default function (
-  callee: Node,
-  thisNode: Node,
-  args: Node[],
+export default function optimiseCallExpression(
+  callee: Expression,
+  thisNode: Expression,
+  args: Readonly<Array<Expression | SpreadElement>>,
```

Comment:
Q: Does TS check that the JSDoc comment types are correct, when using type annotations?

Reply:
I don't think so. The JSDoc is checked only when TypeScript is checking JavaScript files: https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html